corehq/form_processor/backends/couch/update_strategy.py

```diff
@@ -315,20 +315,18 @@ def _apply_attachments_action(self, attachment_action, xform=None):
         """
         # the actions and _attachment must be added before the first saves can happen
         # todo attach cached attachment info
-        def fetch_attachment(name):
-            if xform and 'data' in xform._attachments[name]:
-                assert xform.form_id == attachment_action.xform_id
-                return base64.b64decode(xform._attachments[name]['data'])
-            else:
-                return XFormInstance.get_db().fetch_attachment(attachment_action.xform_id, name)
+        def fetch_attachment(name, _form=[]):
```

Comment:
Clever though confusing use of a mutable argument.

Also if `xform` is not None couldn't you just use that to get the attachment?